app/controllers/account/analytics_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Account::AnalyticsController < ApplicationController
+  before_action :authenticate_user!
+  layout 'account'
+
+  def index
+    redirect_to root_path unless current_user.admin?
```

Comment:
use before_action ....
we have already something for admins

Reply:
Done.